src/Compilers/CSharp/Portable/Symbols/Synthesized/RefKindVector.cs

```diff
@@ -69,5 +76,57 @@ public override int GetHashCode()
         {
             return _bits.GetHashCode();
         }
+
+        public string ToRefKindString()
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
+            var builder = pooledBuilder.Builder;
+
+            builder.Append("{");
+
+            int i = 0;
+            foreach (int byRefIndex in this.Words())
+            {
+                if (i > 0)
+                {
+                    builder.Append(",");
+                }
+
+                builder.AppendFormat("{0:x8}", byRefIndex);
+                i++;
+            }
+
+            builder.Append("}");
+            Debug.Assert(i > 0);
+
+            return pooledBuilder.ToStringAndFree();
+        }
+
+        public static RefKindVector Parse(string refKindString, int capacity)
+        {
+            int index = 0;
+            // To avoid having to map ref kinds again we can just deserialize the BitVector which is what
+            // would have produced the string in the method above
+            var bitVector = BitVector.Create(capacity * 2);
+            foreach (var word in refKindString.Split(','))
+            {
+                var value = Convert.ToUInt64(word, 16);
```

Comment:
>ToUInt64 [](http://example.com/codeflow?start=36&length=8)

May throw on invalid string value

Reply:
What would you rather happen in this case though? I wouldn't think we'd want to ignore the exception, as then we'd emit a duplicate delegate (and it would fail the debug assert at the end of this method too).